scripts/produce_validation_evidence.py

```
@@ -1703,19 +1728,37 @@ def _workflow_catalog_check(
+            if canonical:
+                extent = record.get("implementation_extent")
+                available_operation_ids = record.get("available_operation_ids")
+                if (
+                    extent not in {"implemented", "partial", "planned"}
+                    or not isinstance(available_operation_ids, list)
+                    or not all(
+                        isinstance(operation_id, str)
+                        for operation_id in available_operation_ids
+                    )
+                ):
+                    raise ValueError("workflow contract extent is invalid")
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)